### PR TITLE
Allow demote SRTP call to RTP when using re-INVITE with PJSUA_CALL_REINIT_MEDIA

### PR DESCRIPTION
**Pull request by trengginas**
Although the standards doesn't prohibit this, currently demoting a SRTP call to RTP call is not allowed when `PJSUA_CALL_REINIT_MEDIA` is specified.
This PR will allow demoting such scenario provided that `PJSUA_CALL_REINIT_MEDIA` is used when modifying the call session (using re-INVITE or UPDATE). 

Upon testing, a bug is also found when using DTLS-SRTP with this scenario:
1. Caller send INVITE
2. Callee send 200/OK
3. Caller send re-INVITE with modified media transport address (`PJSUA_CALL_REINIT_MEDIA` is specified). 
4. Callee doesn't reinitiate DTLS renegotiation triggering `PJMEDIA_SRTP_EKEYNOTREADY` error, since on the caller side it has re-initialize the media.

### CODE REVIEW

File: pjmedia/src/pjmedia/transport_srtp_dtls.c [hunk 1]
```diff
@@ -1247,12 +1249,31 @@ static pj_status_t dtls_encode_sdp( pjmedia_transport *tp,
 		    (ds->setup==DTLS_SETUP_ACTIVE? &ID_ACTIVE:&ID_PASSIVE));
 	pjmedia_sdp_media_add_attr(m_loc, a);
 
+        if (last_setup != DTLS_SETUP_UNKNOWN && sdp_remote) {
+            pj_sockaddr rem_rtp;
+            pj_sockaddr rem_rtcp;
+
+            pj_bzero(&rem_rtp, sizeof(rem_rtp));
+            pj_bzero(&rem_rtcp, sizeof(rem_rtcp));
```
Comment (nanangizz) [inline pjmedia/src/pjmedia/transport_srtp_dtls.c:1257]: unnecessary? `pj_sockaddr_init()` will be used in `get_rem_addrs()`

File: pjsip/src/pjsua-lib/pjsua_media.c [hunk 2]
```diff
@@ -3001,6 +3001,7 @@ pj_status_t pjsua_media_channel_deinit(pjsua_call_id call_id)
 	    call_med->tp = call_med->tp_orig = NULL;
 	}
         call_med->tp_orig = NULL;
+        call_med->rem_srtp_use = PJ_FALSE;
```
Comment (nanangizz) [inline pjsip/src/pjsua-lib/pjsua_media.c:3004]: should be set to `PJMEDIA_SRTP_DISABLED`? or create new enum `UNKNOWN` (==`DISABLED`) & set to this.

File: pjmedia/src/pjmedia/transport_srtp_dtls.c [hunk 3]
```diff
@@ -1492,8 +1515,10 @@ static pj_status_t dtls_media_start( pjmedia_transport *tp,
...
 	    /* Attach ourselves to member transport for DTLS nego. */
-	    if (!pj_sockaddr_has_addr(&ds->rem_addr))
-		get_rem_addrs(ds, sdp_remote, media_index);
+            if (!pj_sockaddr_has_addr(&ds->rem_addr)) {
+                get_rem_addrs(ds, sdp_remote, media_index, &ds->rem_addr,
+                              &ds->rem_rtcp);
+            }
 
 	    if (pj_sockaddr_has_addr(&ds->rem_addr))
 		pj_sockaddr_cp(&ap.rem_addr, &ds->rem_addr);
```
Comment (nanangizz) [inline pjmedia/src/pjmedia/transport_srtp_dtls.c:1524]: for consistency, shouldn't we also set `peer_use` (to `MANDATORY`) in `media_start()`?

File: pjmedia/src/pjmedia/transport_srtp_dtls.c (hunk 1 above)
Comment (trengginas) [reply pjmedia/src/pjmedia/transport_srtp_dtls.c]: ok

File: pjmedia/src/pjmedia/transport_srtp_dtls.c (hunk 3 above)
Comment (trengginas) [reply pjmedia/src/pjmedia/transport_srtp_dtls.c:1524]: ok

File: pjsip/src/pjsua-lib/pjsua_media.c (hunk 2 above)
Comment (trengginas) [reply pjsip/src/pjsua-lib/pjsua_media.c]: ok

File: pjmedia/src/pjmedia/transport_srtp_dtls.c [hunk 4]
```diff
@@ -1268,7 +1265,7 @@ static pj_status_t dtls_encode_sdp( pjmedia_transport *tp,
         }
 
 	/* Check if remote signals DTLS re-nego by changing its
-	 * setup/fingerprint in SDP.
+	 * setup/fingerprint in SDP or changed in media transport address.
```
Comment (nanangizz) [inline pjmedia/src/pjmedia/transport_srtp_dtls.c:1268]: perhaps "... setup/fingerprint or media transport address in SDP"?

**Comment by sauwming** [comment]
Make sure it passes pjsua test first before merging since I saw several failures in srtp tests

**Comment by trengginas** [comment]
> Make sure it passes pjsua test first before merging since I saw several failures in srtp tests

Test it here using and ubuntu and passed the tests. Tried testing from a fork with the patch, passed the test.

**Pull request merged**